### PR TITLE
Added back end functionality to remove comment

### PR DESCRIPTION
**Pull request by AirwreckEye**

### CODE REVIEW

File: src/main/java/com/google/ehub/servlets/ItemPageServlet.java [hunk 1]
```diff
@@ -88,4 +88,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       }
     }
   }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response)
+      throws IOException {
+    long commentId;
+    CommentDataManager comment = new CommentDataManager();
```
Comment (codewithjess) [inline src/main/java/com/google/ehub/servlets/ItemPageServlet.java:96]: nit: you can move this line closer to when CommentDataManager is actually used, on line 103.

File: src/main/java/com/google/ehub/servlets/ItemPageServlet.java [hunk 2]
```diff
@@ -88,4 +88,18 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+    try {
+      commentId = Long.parseLong(request.getParameter("commentId"));
+    } catch (NumberFormatException e) {
+      System.err.println("Can't parse itemId to a long");
```
Comment (codewithjess) [inline src/main/java/com/google/ehub/servlets/ItemPageServlet.java:100]: prefer to include the exception in the print statement, which can have more details and make it easier to debug.

**Comment by BryanTriana** [comment]
LGTM after adding tests

File: src/main/webapp/itempagescript.js [hunk 3]
```diff
@@ -124,32 +101,46 @@ function getItemPageComments(comments) {
...
+
+
 /**
  * Creates list element from given comment
  *
- * @param { string } comment - the comment including the username, message, and timestamp
- * @param { boolean } belongsToUser - boolean that shows whether or not the current user has posted a comment
+ * @param { string } comment - the comment including the username, message, and
+ *     timestamp
+ * @param { boolean } belongsToUser - boolean that shows whether or not the
```
Comment (rharagutchi) [inline src/main/webapp/itempagescript.js:133]: missing @param for commentId

File: src/test/java/com/google/ehub/servlets/ItemPageServletTest.java [hunk 4]
```diff
@@ -116,13 +118,81 @@ public void doGetReturnsItemPageData() throws IOException {
...
+    servlet.doGet(request, response);
+    CommentData comment = new CommentData(
+        itemId.getId(), COMMENT, TIMESTAMP, USERNAME, commentId.getId(), belongsToUser);
+    Optional<EntertainmentItem> expectedItem =
+        EntertainmentItemDatastore.getInstance().queryItem(itemId.getId());
+    ItemPageData itemData = new ItemPageData(expectedItem.get(), Lists.newArrayList(comment));
+    CommentData test_comment = commentDataManager.retrieveComments(12345).get(0);
+    verify(response).setContentType(JSON_CONTENT_TYPE);
+    verify(printWriter).println(new Gson().toJson(itemData));
+    Assert.assertEquals(false, test_comment.belongsToUser);
```
Comment (BryanTriana) [inline src/test/java/com/google/ehub/servlets/ItemPageServletTest.java:148]: Adding some spaces between lines would probably improve readability. In the case of this test I think you could add an empty line between sections that setup the test, call the function to be tested, and assert that the results are valid.
Kind of like this:
```suggestion
    Key itemId = EntertainmentItemDatastore.getInstance().addItemToDatastore(selectedItem);
    when(request.getParameter("itemId")).thenReturn(itemId.getId() + "");
    when(response.getWriter()).thenReturn(printWriter);
     Key commentId = commentDataManager.addItemComment(itemId.getId(), COMMENT, TIMESTAMP, ALT_EMAIL);
    
    servlet.doGet(request, response);
    
    CommentData comment = new CommentData(
        itemId.getId(), COMMENT, TIMESTAMP, USERNAME, commentId.getId(), belongsToUser);
    Optional<EntertainmentItem> expectedItem =
        EntertainmentItemDatastore.getInstance().queryItem(itemId.getId());
    ItemPageData itemData = new ItemPageData(expectedItem.get(), Lists.newArrayList(comment));
    CommentData test_comment = commentDataManager.retrieveComments(12345).get(0);
    
    verify(response).setContentType(JSON_CONTENT_TYPE);
    verify(printWriter).println(new Gson().toJson(itemData));
    Assert.assertEquals(false, test_comment.belongsToUser);
```

File: src/test/java/com/google/ehub/servlets/ItemPageServletTest.java [hunk 5]
```diff
@@ -116,13 +118,81 @@ public void doGetReturnsItemPageData() throws IOException {
...
     verify(response).sendError(anyInt(), any());
   }
 
+  @Test
+  public void doGetRetrievesCommentForLoggedOutUser() throws IOException {
+    helper.setEnvIsLoggedIn(false);
+    when(request.getParameter("itemId")).thenReturn("12345");
+    commentDataManager.addItemComment(12345, COMMENT, TIMESTAMP, EMAIL);
+    servlet.doGet(request, response);
+    Assert.assertEquals(1, commentDataManager.retrieveComments(12345).size());
```
Comment (BryanTriana) [inline src/test/java/com/google/ehub/servlets/ItemPageServletTest.java:193]: Same here and empty lines to break different pieces of logic.

File: src/test/java/com/google/ehub/servlets/ItemPageServletTest.java [hunk 6]
```diff
@@ -77,9 +78,10 @@
   @Before
   public void init() {
     MockitoAnnotations.initMocks(this);
-    helper.setUp();
     ProfileDatastore profile = new ProfileDatastore();
     profile.addUserProfileToDatastore("Eric", EMAIL, USERNAME, "Hey");
+    helper.setUp();
+
```
Comment (rharagutchi) [inline src/test/java/com/google/ehub/servlets/ItemPageServletTest.java:84]: Is it the same error when it looks like this? 
```suggestion
public void init() {
    MockitoAnnotations.initMocks(this);
    helper.setUp();	
    ProfileDatastore profile = new ProfileDatastore();
    profile.addUserProfileToDatastore("Eric", EMAIL, USERNAME, "Hey");
```

File: src/test/java/com/google/ehub/servlets/ItemPageServletTest.java (hunk 4 above)
Comment (codewithjess) [inline src/test/java/com/google/ehub/servlets/ItemPageServletTest.java:148]:
```suggestion
    Assert.assertFalse(test_comment.belongsToUser);
```

File: src/test/java/com/google/ehub/servlets/ItemPageServletTest.java [hunk 7]
```diff
@@ -135,17 +139,19 @@ public void doGetReturnsItemPageData_differentUser() throws IOException {
...
+    servlet.doGet(request, response);
     Optional<EntertainmentItem> expectedItem =
         EntertainmentItemDatastore.getInstance().queryItem(itemId.getId());
     ItemPageData itemData = new ItemPageData(expectedItem.get(), Lists.newArrayList(comment));
-    CommentData test_comment = commentDataManager.retrieveComments(12345).get(0);
+
     verify(response).setContentType(JSON_CONTENT_TYPE);
     verify(printWriter).println(new Gson().toJson(itemData));
-    Assert.assertEquals(false, test_comment.belongsToUser);
+    Assert.assertEquals(false, comment.belongsToUser);
```
Comment (codewithjess) [inline src/test/java/com/google/ehub/servlets/ItemPageServletTest.java:154]:
```suggestion
    Assert.assertFalse(comment.belongsToUser);
```

**Pull request merged**